sklearn/ensemble/_iforest.py

```diff
@@ -485,15 +508,28 @@ def _average_path_length(n_samples_leaf):
     n_samples_leaf = n_samples_leaf.reshape((1, -1))
     average_path_length = np.zeros(n_samples_leaf.shape)
 
-    mask_1 = n_samples_leaf <= 1
-    mask_2 = n_samples_leaf == 2
-    not_mask = ~np.logical_or(mask_1, mask_2)
+    mask_small = n_samples_leaf < 52
+    not_mask = ~mask_small
+
+    average_path_length[mask_small] = _average_path_length_small[
+        n_samples_leaf[mask_small]]
+
+    # Average path length equals 2*(H(n)-1), with H(n) the nth harmonic number.
```

Comment:
maybe put this in a `Notes` section of the function dosctring?

Reply:
Great suggestion, I tried to put in a note and a reference. All tests pass, but maybe one of you could check the doocumentation anyway, since for some strange reason Sphinx is crashing on my machine, so I couldn't look at the result.

Reply:
I don't think sphinx render the docstrings of private functions.

Reply:
are you suggesting we move it to the top level description? To me, this is a rather minute detail, so I wouldn't expect someone looking to use the isolation forest to be interested in the implementation of the harmonic number calculation...

Reply:
No this is fine here.